### PR TITLE
TupleFilesystemStoreBackend now ignore `.ipynb_checkpoints` directories

### LINKED ISSUE #1234: Building of HTML Documents raises exceptions (0.9.7)
All,

This is the first time we've tried to build documents on 0.9.7 but we receive an exception when running the context.build_docs(). I have also tried to do this from CLI with 'great_expectations build docs' with same exception.

Initially I thought I had messed up my folder structure locally, so I completely rebuilt doing the following:

1. great_expectations init (CLI)
2. Created a simple expectation suite:
         expectation_suite_name = 'tplt_inbound.transactions.failure'
         context.create_expectation_suite(expectation_suite_name, overwrite_existing=False)

3. Loaded a batch of data from a DF and Added a simple expectation:
         my_batch.expect_column_values_to_be_in_set('ISO_ALPH2_CNTRY_CD', ['CN', 'ID'])
4. Validated the batch:
         my_batch.validate()
5. Saved the Suite:
         my_batch.save_expectation_suite()
6. Using my initial context - tried to build my docs:
         context.build_data_docs()

Presented with the following error:

image

From CLI, error is similar - with more traceback:

image

### PR DESCRIPTION
**Pull request by Aylr**
Fixes #1234

### CODE REVIEW

**Review by jcampbell — CHANGES_REQUESTED** [review]
Two comments here:

1. I think that if we're going to add a mechanism to ignore certain files it should be done via default value and configurability in spirit of others, so that the `IGNORED_FILES` in this PR becomes the default for a configurable status.

2. This default makes sense to me, but with three caveats:
  a) are we protecting against a common misconfiguration, in which case we should try to address the root cause? There "shouldn't" be a case where someone is dropping unexpected files inside the expectation store, so I think this is mostly about a case where someone inadvertently used jupyter notebook to open a suite. In that case, it seems like we should perhaps print a warning about this found key in addition to skipping it.
 b) Current behavior skips the name anywhere in the tuple. Is that always right?
 c) Are there other defaults to drop (.DS_STORE)?

**Comment by Aylr** [comment]
Good thoughts.
1. I'm trying to be as lightweight as possible here, and we can always change this. I specifically don't want to encourage configuration as this is a sacrosanct folder that should not have junk in it, and configuration might encourage this behavior.
2a. We are protecting against users firing up jupyter in expectations folders. I suspect it will be common given how many of our users use notebooks.
2b. Yes.
2c. Maybe, but again, I don't want to encourage this, so I'd opt to leave it as is.

**Review by jcampbell — DISMISSED** [review]

**Comment by Aylr** [comment]
@jcampbell I want you to look at this again - while using this branch I found and fixed another error.

File: great_expectations/data_context/data_context.py [hunk 1]
```diff
@@ -852,6 +852,9 @@ def _compile_evaluation_parameter_dependencies(self):
         self._evaluation_parameter_dependencies = {}
         for key in self.stores[self.expectations_store_name].list_keys():
             expectation_suite = self.stores[self.expectations_store_name].get(key)
+            if not expectation_suite:
+                continue
```
Comment (jcampbell) [inline great_expectations/data_context/data_context.py:856]: I liked the earlier version that had a warning here. But I think we can include that in updates to the store API later.

**Review by jcampbell — APPROVED** [review]

**Comment by Aylr** [comment]
Merging. Required builds passed: https://travis-ci.org/github/great-expectations/great_expectations/builds/672699929

**Pull request merged**